### ISSUE TITLE
[Feature Request] Soft delete feature

### ISSUE THREAD

**Issue by devsamt** [getinsomnia/insomnia#3268]
I accidentally deleted a workspace after a UI update I'm not used to. Please, can you implement a soft delete behavior in order to be able to recover deleted workspace in the event of improper handling ? I think deleted workspace can be hard deleted after 30 days or on user confirmation (in a trash container).

Illustration:
![2021-04-07 17_13_20-Éléments envoyés - a gragnano@samt fr - Outlook](https://user-images.githubusercontent.com/53750216/113890326-57f52580-97c4-11eb-8ce1-c9027a6149c4.png)

**Comment by dimitropoulos** [comment]
Could you please say a little more about the use-case.

You said "improper handling" but it's _not_ one of accidental deletion, right?  I ask because we already have a confirmation dialog for that:

![Screenshot_20210407_141712](https://user-images.githubusercontent.com/15232461/113914804-f32dd100-97ab-11eb-899f-6770e57d017b.png)

What kind of improper handling are you referring to?

**Comment by devsamt** [comment]
After application upgrade, some "default" workspaces have been automatically added ("Insomnia" and "Insomnia Designer"). I decided to delete both because empty and unused. Instinctively, I memorized the positions of the elements to be deleted and proceeded to delete these two "positions" ... but after the first deletion the remaining elements were reorganized. This is a stupid and very damaging handling error given the work already done on this workspace.
